### PR TITLE
sessionctx/variable: remove dependency on context package.

### PR DESCRIPTION
**Pull request by coocood**
Makes using session variable easier and faster.
Prepare for use session variable to handle data truncated error.

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @zimulala PTAL

File: bootstrap_test.go [hunk 1]
```diff
@@ -88,18 +86,16 @@ func (s *testSessionSuite) TestBootstrap(c *C) {
...
+		parser:      parser.New(),
+		sessionVars: variable.NewSessionVars(),
 	}
 	ss.SetValue(context.Initing, true)
 	domain, err := domap.Get(store)
 	c.Assert(err, IsNil)
 	sessionctx.BindDomain(ss, domain)
-	variable.BindSessionVars(ss)
-	variable.GetSessionVars(ss).SetStatusFlag(mysql.ServerStatusAutocommit, true)
 	// session implements autocommit.Checker. Bind it to ctx
```
Comment (shenli) [inline bootstrap_test.go]: Remove the comment.

**Comment by shenli** [comment]
LGTM

File: ddl/reorg.go [hunk 2]
```diff
@@ -102,12 +105,17 @@ func (c *mockContext) ClearValue(key fmt.Stringer) {
...
 func (d *ddl) newMockContext() context.Context {
 	c := &mockContext{
-		store: d.store,
-		m:     make(map[fmt.Stringer]interface{}),
+		store:       d.store,
+		m:           make(map[fmt.Stringer]interface{}),
+		sessionVars: variable.NewSessionVars(),
 	}
-
+	c.sessionVars.SetStatusFlag(mysql.ServerStatusAutocommit, false)
```
Comment (hanfei1991) [inline ddl/reorg.go]: Why set it false ?

File: evaluator/builtin_info.go [hunk 3]
```diff
@@ -21,14 +21,12 @@ import (
...
 	"github.com/pingcap/tidb/mysql"
-	"github.com/pingcap/tidb/sessionctx/db"
-	"github.com/pingcap/tidb/sessionctx/variable"
 	"github.com/pingcap/tidb/util/types"
 )
 
 // See https://dev.mysql.com/doc/refman/5.7/en/information-functions.html
 func builtinDatabase(args []types.Datum, ctx context.Context) (d types.Datum, err error) {
-	s := db.GetCurrentSchema(ctx)
+	s := ctx.GetSessionVars().CurrentDB
```
Comment (hanfei1991) [inline evaluator/builtin_info.go]: maybe s is not a good name.

**Comment by zimulala** [comment]
LGTM

File: session.go [hunk 4]
```diff
@@ -605,11 +594,11 @@ func (s *session) GetTxn(forceNew bool) (kv.Transaction, error) {
 		}
 		ac = s.isAutocommit(s)
 		if !ac {
-			variable.GetSessionVars(s).SetStatusFlag(mysql.ServerStatusInTrans, true)
+			s.sessionVars.SetStatusFlag(mysql.ServerStatusInTrans, true)
```
Comment (hanfei1991) [inline session.go]: Can we move the line 581-584 and line 595-598 out of brackets?

File: ddl/reorg.go (hunk 2 above)
Comment (coocood) [inline ddl/reorg.go]: It was false before, but `NewSessionVars` set autocommit to true.
Set it to false so the test works the same as before.

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**